cli/src/commands/create/integrations.rs

```diff
@@ -0,0 +1,100 @@
+use anyhow::{bail, Context, Result};
+use colored::Colorize;
+use dialoguer::Confirm;
+use std::path::PathBuf;
+
+use log::info;
+use reinfer_client::{
+    resources::integration::{Integration, NewIntegration},
+    Client, IntegrationFullName,
+};
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub struct CreateIntegrationArgs {
+    #[structopt(short = "f", long = "file", parse(from_os_str))]
+    /// Path to JSON file with integration
+    path: PathBuf,
+
+    #[structopt(name = "name")]
+    /// Name of the new integration
+    name: IntegrationFullName,
+
+    #[structopt(long)]
+    /// Whether to overwrite an existing integration with the same name
+    overwrite: bool,
+}
+
+pub fn create(client: &Client, args: &CreateIntegrationArgs) -> Result<()> {
+    let CreateIntegrationArgs {
+        path,
+        name,
+        overwrite,
+    } = args;
+
+    let new_integration = read_integration(path)?;
+
+    let mut integrations = client.get_integrations()?;
+
+    integrations
+        .retain(|integration| format!("{}/{}", integration.owner.0, integration.name.0) == name.0);
+
+    if let Some(existing) = integrations.first() {
+        if *overwrite {
+            overwrite_integration(client, name, &new_integration, existing)?;
+            info!("Updated integration {}", name.0);
+        } else {
+            bail!("Provide the `--overwrite` flag to update an existing integration")
+        }
+    } else {
+        client.put_integration(name, &new_integration)?;
+        info!("Created new integration {}", name.0);
+    }
+
+    Ok(())
+}
+
+fn overwrite_integration(
+    client: &Client,
+    name: &IntegrationFullName,
+    new_integration: &NewIntegration,
+    old_integration: &Integration,
+) -> Result<()> {
+    let old_integration: NewIntegration =
+        serde_json::from_str(&serde_json::to_string(old_integration)?)?;
```

Comment:
Not great, not terrible :shrug: you could have implemented `From` but this does the job.